### PR TITLE
feat: Auto rotate site bucket keys #153

### PR DESCRIPTION
**Pull request by apburnes**
## Changes proposed in this pull request:
- Adds CI task to rotate keys nightly
- Tests key rotation
- Updates to node v20 to for built in env file consumption

## security considerations
Rotates Pages site bucket keys in small batches nightly. The platform rotates all site bucket keys  roughly every 30 days.
Updates to node v20

### CODE REVIEW

File: ci/pipeline-staging.yml [hunk 1]
```diff
@@ -461,6 +461,51 @@ jobs:
             username: ((slack-username))
             icon_url: ((slack-icon-url))
 
+  - name: nightly-site-bucket-key-rotator
+    plan:
+      - get: src
+        resource: pr-((deploy-env))
```
Comment (drewbo) [inline ci/pipeline-staging.yml:467]: staging's pr resource is just named `pr` (since it runs on release PRs and dev PRs)

**Review by drewbo — CHANGES_REQUESTED** [review]
- Flagged two gotchas in the ci pipelines (which are the result of my deploy changes). 
- I'd move the script from its currently location into `ci/tasks` for consistency, unless you think we'll run it locally
- I'd document somewhere that this is running the 20 most recent
- How easy/helpful would it be to add a test of the service? Optional, only if it adds value (since I imagine it might be heavily reliant on mocks)

File: ci/pipeline-production.yml [hunk 2]
```diff
@@ -386,6 +386,51 @@ jobs:
             channel: ((slack-channel))
             username: ((slack-username))
             icon_url: ((slack-icon-url))
+
+  - name: nightly-site-bucket-key-rotator
+    plan:
+      - get: src
+        resource: pr-((deploy-env))
```
[the remark this answers is not in the document]
Comment (apburnes) [reply ci/pipeline-production.yml]: Updated

File: ci/pipeline-staging.yml (hunk 1 above)
Comment (apburnes) [reply ci/pipeline-staging.yml]: Updated

**Comment by apburnes** [comment]
@drewbo updated based on your last review

File: ci/tasks/get-app-env.sh [hunk 3]
```diff
@@ -7,5 +7,8 @@ cf auth
 
 cf t -o $CF_ORG -s $CF_SPACE
 
-echo "VCAP_SERVICES={`cf env $CF_APP_NAME | tail -n+3 | awk -v RS= 'NR==1' | tail -n+2 | tr -d '\n'`" >> .env
-echo "VCAP_APPLICATION={`cf env $CF_APP_NAME | tail -n+3 | awk -v RS= 'NR==2' | tail -n+2 | tr -d '\n'`" >> .env
+CF_APP_GUID=`cf app $CF_APP_NAME --guid`
+
+cf curl /v3/apps/$CF_APP_GUID/env | \
```
Comment (drewbo) [inline ci/tasks/get-app-env.sh:12]: 😍

**Review by drewbo — APPROVED** [review]

**Review by drewbo — APPROVED** [review]

**Pull request merged**